src/main/java/racing/domain/car/Cars.java

```diff
@@ -0,0 +1,80 @@
+package racing.domain.car;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import racing.domain.startegy.MovingStrategy;
+
+public class Cars {
+
+    private static final int RANGE_OF_RANDOM_NUMBER = 10;
+
+    private final List<Car> cars;
+
+    public Cars(List<Car> cars) {
+        this.cars = cars;
+        checkZeroCar(this.cars);
+        checkSameNameCars(this.cars);
```

Comment:
객체의 상태 값을 초기화하는 것 보다 유효성을 체크하는 로직이 먼저 수행되어야 할 것 같아요! 
```suggestion
        checkZeroCar(cars);
        checkSameNameCars(cars);
        this.cars = cars;
```

또한 만약 이런 식으로 검증해야할 로직이 많다면 생성자가 너무 커지는 문제가 발생할 수 있어요. 👀 
이런 경우에는 아래와 같이 하나의 메서드에 유효성 로직들을 모두 담은 뒤, 생성자에는 하나의 메서드만 호출하도록 변경해볼 수 있어요! 😃

```java
    public Cars(List<Car> cars) {
        validateCars(cars);
        this.cars = cars;
    }

    private void validateCars(List<Car> cars) {
        checkZeroCar(cars);
        checkSameNameCars(cars);
    }
```

Reply:
앗 넵! 체크하는 로직을 먼저 수행한 후, 상태값을 초기화하겠습니다!
확실히 validateCars라는 메서드 내부에 여러 개의 유효성 검사를 두는 것이 생성자의 크기가 줄어드네요! 
감사합니다!